sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -150,6 +150,20 @@ trait CheckAnalysis extends PredicateHelper {
           case s: SubqueryExpression =>
             checkSubqueryExpression(operator, s)
             s
+
+          case a @ AnySubquery(values, _, genCmp) =>
+            // ANY/SOME predicate doesn't support multi-column comparison.
+            if (values.size > 1) {
+              val cmp = genCmp(null, null)
+              cmp match {
+                case EqualTo(_, _) => // Ok
+                  a
+                case other =>
+                  failAnalysis("ANY/SOME predicate doesn't support multi-column comparison.")
```

Comment:
@francis0407 Isn't EqualTo considered a comparison operation ? Should we include the exact comparison operation in the error text ? Like : 
"ANY/SOME predicate does not support '>" operation when multiple columns are specified as its operands. 
.. or something like that ..